docs: update redirects doc

vinckr opened the pull request:
Make the redirection document more clear as described here:
closes https://github.com/ory/cloud/issues/51
Also revised the entire document

## Related Issue or Design Document

<!--
If this pull request

1. is a fix for a known bug, link the issue where the bug was reported in the format of `#1234`;
2. is a fix for a previously unknown bug, explain the bug and how to reproduce it in this pull request;
2. implements a new feature, link the issue containing the design document in the format of `#1234`;
3. improves the documentation, no issue reference is required.

Pull requests introducing new features, which do not have a design document linked are more likely to be rejected and take on average 2-8 weeks longer to
get merged.

You can discuss changes with maintainers either in the Github Discusssions in this repository or
join the [Ory Chat](https://www.ory.sh/chat).
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
      and signed the CLA.
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [x] I have added necessary documentation within the code base (if
      appropriate).

## Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

tomekpapiernik commented on docs/concepts/redirects.mdx:
```diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
-below the https://sub.domain.myapp.com/only/path needs to match perfectly for
-this sub-domain and path for it to be allowed. Other redirects using myapp.com
-will fail, however, for https://anotherapp.com will succeed on any path.
+parameter on [Self-Service Flows](./self-service.mdx). For example: A user opens
```
```suggestion
parameter on [Self-Service Flows](./self-service.mdx). For example: a user opens
```

tomekpapiernik commented on docs/concepts/redirects.mdx:
```diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
+parameter on [Self-Service Flows](./self-service.mdx). For example: A user opens
+a sharable link to go to `https://myapp.com/posts`. This URL requires the user
+to be signed in and redirects the user back to the login page. To return the
+user back to the original URL append `?return_to=https://myapp.com/posts` when
```
```suggestion
user back to the original URL, append `?return_to=https://myapp.com/posts` when
```

tomekpapiernik commented on docs/concepts/redirects.mdx:
```diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
+### Login, registration and settings
 
-And Settings has three fields (all optional):
+The `Login` and `Registration` flows have three fields (all optional):
```
Please get rid of the code formatting for Login and Registration. Previously the flow names were in bold so let's be consistent and bold them here, too. (Not a fan of that too, but at least we'll be consistent ;) )

tomekpapiernik commented on docs/concepts/redirects.mdx:
```diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
-2. An `Post-Password Redirect URL`
-3. An `Post-Profile Redirect URL`
+2. `Post-Password Redirect URL`
+3. `Post-OIDC Redirect URL` (_coming soon_)
```
No "coming soon" in the docs please. Docs should describe the current state of the software.

tomekpapiernik commented on docs/concepts/redirects.mdx:
```diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
 ![Browser Redirects URL Management Transformed](/images/browser-redirects/browser-redirects-url-management-2.png)
 
-### Verification, Recovery and Logout
+### Verification, recovery and logout
```
```suggestion
### Verification, recovery, and logout
```

tomekpapiernik commented on docs/concepts/redirects.mdx:
````diff
@@ -12,133 +12,156 @@ to configure it on Ory Cloud console. If you are interested in utilizing the SDK
+```
 
-### Post Flow
+### `incorrect format`
````
Let's avoid code in headings. Sometimes you just can't, but here your heading is JUST `code`. Try to think of a better one ;)

vinckr commented:
Need to fix a few things, ETA 30 min.

aeneasr commented on docs/concepts/redirects.mdx:
````diff
@@ -3,35 +3,33 @@ id: redirects
+starting the self-service login flow `initSelfServiceLogin`:
+
+```js
+    "request_url": "http://<your-project>.projects.oryapis.com/self-service/login/browser?return_to=...",
````
This is confusing, `initSelfServiceLogin`  does not accept the aprameter like this. It also depends on the SDK that you're using. Maybe it makes more sense to link to the API doc here:

https://www.ory.sh/docs/reference/api#operation/submitSelfServiceLoginFlow

and show an example with cURL:

```suggestion
curl -X GET 'http://<your-project>.projects.oryapis.com/self-service/login/browser?return_to=...'
```

aeneasr commented on docs/concepts/redirects.mdx:
```diff
@@ -40,105 +38,125 @@ will fail, however, for https://anotherapp.com will succeed on any path.
+2. `Post-Password Redirect URL`
+3. `Post-Profile Redirect URL`
+
+To overwrite the `Global Redirect URL` set the `Default Redirect URL` to a fully
+qualified domain name
+([FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) - for example
+`https://domain.example/login`).  
+To replace the `Global Redirect URL` path set the `Default Redirect URL` to a
+path. For example when setting it to `/login`,
+`https://domain.example/global/path` becomes `https://domain.example/login`.
```
I don't think this works any more, can you confirm that this is true? I think you now just need to always specify the FQDN if it's not your CNAME or hosted URL or Ory Proxy base URL

aeneasr commented:
Also, I have to say that I find this quite difficult to read, because there are so many code boxes:

<img width="1065" alt="Bildschirmfoto 2022-03-18 um 14 26 57" src="https://user-images.githubusercontent.com/3372410/159011237-c5ec5af2-eb47-4623-bac1-dc9676788c72.png">

Can there be maybe a bit better format? Code boxes should really only be used for code.

Another question is whether it would make sense to add a mermaid graph explaining the logic how we choose which method gets used?

aeneasr commented on docs/concepts/redirects.mdx:
````diff
@@ -3,35 +3,33 @@ id: redirects
+starting the [self-service login flow](https://www.ory.sh/docs/reference/api#operation/initializeSelfServiceLoginFlowForBrowsers):
+
+```js
+    curl -X GET 'http://<your-project>.projects.oryapis.com/self-service/login/browser?return_to=...'
````
```suggestion
curl -X GET 'http://<your-project>.projects.oryapis.com/self-service/login/browser?return_to=...'
```

aeneasr commented on docs/concepts/redirects.mdx:
```diff
@@ -40,105 +38,125 @@ will fail, however, for https://anotherapp.com will succeed on any path.
+2. Post-Password Redirect URL
+3. Post-Profile Redirect URL
+
+To overwrite the Global Redirect URL set the Default Redirect URL to a fully
```
I find this a bit confusing, what's the difference between "Global Redirect URL" and "Default Redirect URL"?

vinckr replied:
See here : https://github.com/ory/docs/pull/668/files#r830157781

 I see to clarify this a bit and see how it works without the Global.

aeneasr replied:
Oh I see, so the key in Ory Kratos is always `default_browser_return_url`, and then we have different prefixes for it. I think this is a bit confusing when reading it in the docs. It might be a good idea to present this in a diagram or something - even a YAML example would help.

Maybe this document helps also?

https://www.ory.sh/docs/kratos/concepts/browser-redirect-flow-completion

aeneasr commented on docs/concepts/redirects.mdx:
```diff
@@ -40,105 +38,125 @@ will fail, however, for https://anotherapp.com will succeed on any path.
+qualified domain name
+([FQDN](https://en.wikipedia.org/wiki/Fully_qualified_domain_name) - for example
+`https://domain.example/login`).  
+To replace the Global Redirect URL path set the Default Redirect URL to a
+path. For example when setting it to `/login`,
+`https://domain.example/global/path` becomes `https://domain.example/login`.
```
I don't quite understand what this means? Are you trying to explain that you can set both an absolute (`https://www.example.org/some/path`) and relative (`/some/path`) URL?

aeneasr commented on docs/concepts/redirects.mdx:
```diff
@@ -40,105 +38,125 @@ will fail, however, for https://anotherapp.com will succeed on any path.
+can be set.  
+Set the Post- redirect to a FQDN to replace the closest relative, starting
+with the Login/Registration Default Redirect URL and then the
+Global Redirect URL.  
```
I think we should call just call this `Default Redirect URL` and not introduce `Global`, I think they refer to the same thing?

vinckr replied:
I kind of went with the wording in the original doc - which follows the names in the UI.

However I think it makes sense to omit the Global and call it the Default Redirect URL.

<img width="762" alt="Screenshot 2022-03-18 at 17 13 19" src="https://user-images.githubusercontent.com/55394283/159040821-f0717237-ed4b-47b5-a7ff-654752b7b968.png">

"Global Redirect URL" would be the "global default redirect URL" in the very top 

"Default Redirect URL" means the "Post- Default Redirect URLs", so for example "Post-Login Default Redirect URL".

That is the source of confusion. It is probably sensible to give the Post-X flows a common name to avoid this confusion? Maybe "Default Redirect URLs for Self-Service Flows"?
and then the Default Redirect URL is the global one.

aeneasr replied:
How about "Redirect target after successful <login|registration|...>" and "Redirect target after successful <login|registration|...> with <password|social|...>"?

aeneasr commented on docs/concepts/redirects.mdx:
```diff
@@ -40,105 +38,125 @@ will fail, however, for https://anotherapp.com will succeed on any path.
-URLs matching any of the denied URLs, such as the Ory domain will automatically
-be denied. This deny list can be extended in future to prevent any abuse on the
-Ory Cloud.
+It's not possible to set the `ory` domain as redirect URL.
```
```suggestion
It's not possible to set the any Ory-owned domain as redirect URL.
```

vinckr commented:
I removed the "Global" Redirects term, so its consistently Default Redirects URL and the others are called Post-Login Redirect URL and After Password Redirect URL etc.

I also added the .yml code example with some comments. 

I wanted to have a bigger .yml that works both for OSS and Cloud documentation, but we need to be able to display only specific lines (so we dont have 10 .yml files for this). Maybe it is also a bit overkill.

aeneasr commented on docs/concepts/redirects.mdx:
````diff
@@ -40,105 +38,148 @@ will fail, however, for https://anotherapp.com will succeed on any path.
+3. Post-OIDC Redirect URL
+
+```yaml file="kratos.config.yml"
+selfservice:
````
As a follow up we should show how to set this using tabs:

1. Self-Hosted Kratos (this config file here)
2. Ory Console (screenshot)
3. Ory CLI (`ory patch project .. --replace /.../...`)

I'm ok not doing this in this PR but can you please create a follow up issue and tackle this soon? Because this section is otherwise not very helpful for cloud customers. Please link the issue here

aeneasr reviewed (approved):
Nice!
